### PR TITLE
Features polish

### PR DESCRIPTION
**Pull request by Xaroz**
- Added all provinces (not used rn)
- Added validation for some fields (like document, password and date)
- Added SMS code confirmation, user must register and confirm email, then login and if the user hasn't confirmed his phone he will be sent to phone validation page
- Added create client

KNOWN Issues:
- Needs to refactor and make the code more readable

### CODE REVIEW

**Comment by Xaroz** [comment]
@Evalir check the second known issue if possible please

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+
+import { getDateFromString, validatePhoneNumber } from '../../utils'
+import {
+  CONFIRMATION_CODE_LENGTH,
+  PHONE_NUMBER_LENGTH,
+} from '../../constants/commons'
+import { VALIDATION_MESSAGES } from '../../constants/errors'
+import axios from 'axios'
+
+//TODO:
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:20]: Remove this.

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+  >('')
+
+  const [confirmationCode, setConfirmationCode] = useState<string>('')
+  const [confirmationCodeError, setConfirmationCodeError] = useState<string>('')
+  const [validationId, setValidationId] = useState<string>('')
+
+  const [token, setToken] = useState<string>('')
+  const [step, setStep] = useState<number>(1)
+
+  // const history = useHistory()
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:47]: Remove this.

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+        validatePhoneNumber(phoneNumber.substring(2))
+      const isDateValid = date.length >= 1
+
+      if (isNameValid) setNameErrorMessage('')
+      if (isDocumentValid) setDocumentErrorMessage('')
+      if (isPhoneNumberValid) setPhoneNumberErrorMessage('')
+      if (isDateValid) setDateErrorMessage('')
+
+      if (isNameValid && isDocumentValid && isPhoneNumberValid && isDateValid) {
+        // handleNextStep()
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:86]: Remove this.

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+        const config = {
+          headers: {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            Authorization: token,
+          },
+        }
+
+        try {
+          const result = await axios.post(url, body, config)
+          console.log('PERFORM RESULT: ', result)
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:167]: Remove this

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+      body.append('name', name)
+      body.append('document', document)
+      body.append(
+        'birthdate',
+        `${newDate.year}-${newDate.monthNumber}-${newDate.day}`
+      )
+      body.append('phone_number', phoneNumber)
+      const url = `${process.env.REACT_APP_API_BASE_URL}${path}`
+      console.log(url)
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:207]: Remove this

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+      const config = {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+          Authorization: token,
+        },
+      }
+
+      try {
+        const result = await axios.post(url, body, config)
+        console.log('CREATE CLIENT RESULT ', result)
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:218]: Remove this.

File: src/pages/PhoneValidationPage/PhoneValidationPage.tsx [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+  )
+
+  useEffect(() => {
+    async function getToken() {
+      try {
+        const tokenResult = `Bearer ${(await Auth.currentSession())
+          .getIdToken()
+          .getJwtToken()}`
+        setToken(tokenResult)
+        console.log('GET TOKEN', tokenResult)
```
Comment (PseudoMera) [inline src/pages/PhoneValidationPage/PhoneValidationPage.tsx:233]: Remove this.

File: src/pages/RegisterPage/RegisterPage.tsx [hunk 8]
```diff
@@ -75,98 +49,33 @@ const RegisterPage: React.FC<RegisterPageProps> = ({ history }) => {
...
         try {
           const { user } = await Auth.signUp({
             username: email,
             password: password,
             attributes: {
               email: email,
-              phone_number: phoneNumber,
             },
           })
+          console.log('USER REGISTER', user)
```
Comment (PseudoMera) [inline src/pages/RegisterPage/RegisterPage.tsx:71]: Remove this.

File: src/pages/RegisterPage/RegisterPage.tsx [hunk 9]
```diff
@@ -176,84 +85,70 @@ const RegisterPage: React.FC<RegisterPageProps> = ({ history }) => {
...
+    [handleNextStep]
   )
 
   const handleConfirmCodeSubmit = useCallback(async () => {
     try {
       const result = await Auth.confirmSignUp(email, confirmationCode)
-      history.push('/')
-      console.log('BOBO: ', result)
+      history.goBack()
+      console.log('CONFIRMATION RES: ', result)
```
Comment (PseudoMera) [inline src/pages/RegisterPage/RegisterPage.tsx:106]: Remove this.

File: src/pages/RegisterPage/RegisterPage.tsx [hunk 10]
```diff
@@ -176,84 +85,70 @@ const RegisterPage: React.FC<RegisterPageProps> = ({ history }) => {
...
-      console.error('Confirmation error: ', error)
       setConfirmationCodeError('Código incorrecto')
     }
   }, [confirmationCode, email])
 
   const handleResendCodeSubmit = useCallback(async () => {
     try {
       const result = await Auth.resendSignUp(email)
-      console.log('BOBO2: ', result)
+      console.log('RESEND CONFIRMATION CODE: ,', result)
```
Comment (PseudoMera) [inline src/pages/RegisterPage/RegisterPage.tsx:115]: Remove this.

File: src/pages/RegisterPage/steps/Step1.tsx [hunk 11]
```diff
@@ -184,7 +173,7 @@ const Step3: React.FC<Step3Props> = ({
             />
             <p css={errorStyle}>{emailErrorMessage}</p>
 
-            <IonSelect
+            {/* <IonSelect
```
Comment (PseudoMera) [inline src/pages/RegisterPage/steps/Step1.tsx:176]: Remove this.

**Pull request merged**